frontend/components/groups/js/groups.service.ts

```diff
@@ -303,45 +304,78 @@ export class GroupsService {
 	 */
 	public selectGroup(group: any) {
 
+		const sameGroup = this.state.selectedGroup === group;
+		const multi = this.MultiSelectService.isMultiMode();
+
 		if (this.state.selectedGroup) {
 			this.state.selectedGroup.selected = false;
 		}
 		this.state.selectedGroup = group;
 		this.state.selectedGroup.selected = true;
 		this.state.selectedGroup.totalSavedMeshes = 0;
 
-		// if (this.state.selectedGroup.objects && this.state.selectedGroup.objects.length) {
-
-		const multi = this.MultiSelectService.isMultiMode();
 		let color = this.ViewerService.getDefaultHighlightColor();
-
 		if (!this.state.selectedGroup.new) {
 			color = this.state.selectedGroup.color.map((c) => c / 255);
 		}
 
-		// If multi is enabled we want to inverse selections
-		// rather than add them
-		let additive = (multi === true) ? false : true;
+		if (!multi) {
+			this.state.multiSelectedGroups = [group];
+			return this.TreeService.selectNodesBySharedIds(
+				this.state.selectedGroup.objects,
+				multi, // multi
+				color,
+				true,
+			).then((meshes) => {
 
-		if (this.state.selectedGroup.objects.length === 0) {
-			additive = false;
-		}
+				// If we haven't saved don't update saved meshes
+				if (!this.state.selectedGroup.new) {
+					const total = this.getTotalMeshes(meshes);
+					this.state.selectedGroup.totalSavedMeshes = total;
+				}
 
-		return this.TreeService.selectNodesBySharedIds(
-			this.state.selectedGroup.objects,
-			multi, // multi
-			additive,
-			color,
-			false,
-		).then((meshes) => {
-
-			// If we haven't saved don't update saved meshes
-			if (!this.state.selectedGroup.new) {
-				const total = this.getTotalMeshes(meshes);
-				this.state.selectedGroup.totalSavedMeshes = total;
-			}
+			});
 
-		});
+		} else if (!this.state.multiSelectedGroups.includes(group)) {
+
+			// selecting group that's not selected
+			this.state.multiSelectedGroups.push(group);
+
+			return this.TreeService.selectNodesBySharedIds(
```

Comment:
This looks exactly the same as Line 324-337 above. Can we delete some code?

Reply:
I can always delete code @Charence